src/main/java/com/mraof/minestuck/client/gui/RemoteObserverScreen.java

```diff
@@ -66,33 +76,41 @@ private void changeActiveType()
 	}
 	
 	/**
-	 * Returns the current entity type if the text field string matches it and the active type is correct, returns players as a generic entity type if not valid but active type is correct, or returns null if the correct active type isnt selected
+	 * Attempts to increase or decrease the range at which the observer attempts to detect entities
 	 */
-	private EntityType<?> getEntityType(String stringInput)
+	private void changeRange(int change)
 	{
-		if(activeType == RemoteObserverTileEntity.ActiveType.CURRENT_ENTITY_PRESENT)
-		{
-			Optional<EntityType<?>> attemptedEntityType = EntityType.byString(stringInput);
-			return attemptedEntityType.orElse(EntityType.PLAYER);
-		} else
-			return null;
+		observingRange = MathHelper.clamp(observingRange + change, 1, 64);
+		incrementButton.active = observingRange < 64;
+		decrementButton.active = observingRange > 1;
+		largeIncrementButton.active = observingRange < 64;
+		largeDecrementButton.active = observingRange > 1;
 	}
 	
 	@Override
 	public void render(MatrixStack matrixStack, int mouseX, int mouseY, float partialTicks)
 	{
 		this.renderBackground(matrixStack);
 		RenderSystem.color4f(1F, 1F, 1F, 1F);
-		this.minecraft.getTextureManager().bind(guiBackground);
-		int yOffset = (this.height / 2) - (guiHeight / 2);
+		this.minecraft.getTextureManager().bind(GUI_BACKGROUND);
+		int yOffset = (this.height / 2) - (GUI_HEIGHT / 2);
 		
-		this.blit(matrixStack, (this.width / 2) - (guiWidth / 2), yOffset, 0, 0, guiWidth, guiHeight);
+		this.blit(matrixStack, (this.width / 2) - (GUI_WIDTH / 2), yOffset, 0, 0, GUI_WIDTH, GUI_HEIGHT);
+		font.draw(matrixStack, Integer.toString(observingRange), (width / 2) - 5, (height - GUI_HEIGHT) / 2 + 55, 0x404040);
 		super.render(matrixStack, mouseX, mouseY, partialTicks);
 	}
 	
 	private void finish()
 	{
-		MSPacketHandler.sendToServer(new RemoteObserverPacket(activeType, te.getBlockPos(), getEntityType(entityTypeTextField.getValue())));
-		onClose();
+		Optional<EntityType<?>> attemptedEntityType = EntityType.byString(entityTypeTextField.getValue());
+		
+		if(attemptedEntityType.isPresent() && RemoteObserverTileEntity.entityCanBeObserved(attemptedEntityType.get()))
```

Comment:
From my understanding of this screen, the entity type should only matter and be used for the `CURRENT_ENTITY_PRESENT` type, but here you require that there is a valid entity type regardless of which observer type is being set.

Reply:
It can now move forward with an invalid entity type present as long as the active type is not CURRENT_ENTITY_PRESENT. It will stick with the original entity type stored in the tile entity if its invalid but regardless of active type it will move forward with the widget entity type if it passes the original checks